core/src/main/scala/org/apache/spark/h2o/backends/internal/InternalH2OBackend.scala

```diff
@@ -76,6 +89,23 @@ object InternalH2OBackend extends InternalBackendUtils {
     }
   }
 
+  private def lockCloud(endpoints: Array[RpcEndpointRef]): Unit = {
+    endpoints.foreach {
```

Comment:
Do we have send this message to all of them. it would cause any harm, but would the cloud be locked multiple times?

Reply:
It is safe to call cloud lock multiple times, but not neccessary. I will invoke it just on the head of the list